src/ingester/fetchers/grpc.rs

```diff
@@ -78,6 +78,10 @@ pub fn get_grpc_stream_with_rpc_fallback(
                             panic!("gRPC stream ended unexpectedly");
                         }
                         Either::Right((Some(rpc_blocks), _)) => {
+                            let rpc_blocks: Vec<BlockInfo> = rpc_blocks
+                                .into_iter()
+                                .filter(|b| b.metadata.slot > last_indexed_slot)
+                                .collect();
                             let blocks_len = rpc_blocks.len();
                             let parent_slot = rpc_blocks.first().unwrap().metadata.parent_slot;
                             let last_slot = rpc_blocks.last().unwrap().metadata.slot;
```

Comment:
During certain concurrency conditions there is a chance that the first slot in the block batch might be equal or smaller than the last_indexed_slot. If it is, then the parent_slot == last_indexed_slot check fails.